components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/util/AuthenticatedUserBuilder.java

```diff
@@ -179,22 +181,54 @@ private User resolveLocalUserFromUserStore(String userId, AuthenticatedUserData.
         throw new ActionExecutionResponseProcessorException(errorMessage);
     }
 
-    private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse() {
+    private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse()
+            throws ActionExecutionResponseProcessorException {
 
         Map<ClaimMapping, String> userAttributes = new HashMap<>();
         for (AuthenticatedUserData.Claim claim : userFromResponse.getUser().getClaims()) {
+            validateClaimValue(claim.getUri(), claim.getValue());
             if (GROUP_CLAIM.equals(claim.getUri())) {
                 ignoreGroupsInClaimsInResponse();
                 continue;
+            } else if (ROLES_CLAIM.equals(claim.getUri())) {
```

Comment:
we should add a log here that role claim is not supported

Reply:
well I meant a diagnostic log. Else there's no indication to the external developer on what happens to role claim

Reply:
Address the comment